_episodes/02-benchmark-and-scaling.md

```diff
@@ -19,27 +19,244 @@ keypoints:
+> > ~~~
+> > {: .input}
+> {: .solution}
+{: .challenge}
+
+## Understand the output files
+Let us now look for the output files. You would notice that in this case three files have been created: ```log.lammps```, ```mpi-out.xxxxx```, and ```mpi-err.xxxxx```. Among these three, ```mpi-out.xxxxx``` is mainly to capture the screen output that have been generated during the job execution. The purpose of the ```mpi-err.xxxxx``` file is to log entries if there is any error occurring during run-time. The one that is created by LAMMPS is called ```log.lammps```. 
+
+Once you open the ```log.lammps``` file you will notice that this file contains most of the important information starting from the LAMMPS version, number of processors used for the runs, processor lay out, thermdynamic steps, and the timing information. The header of a ```log.lammps``` file would be somewhat similar to this:
+```
+LAMMPS (18 Feb 2020)
+OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:94)
+  using 1 OpenMP thread(s) per MPI task
+```
+Note that it tells you about the LAMMPS version, and ```OMP_NUM_THREADS``` which is one of the important environment variables required to set for obtainging a performance boost. This would be discussed later in this tutorial. But for now, we'll focus mainly on the timing information provided in this file. 
+
+When the run concludes, LAMMPS prints the final thermodynamic state and a total run time for the simulation. It also appends statistics about the CPU time and storage requirements for the simulation. An example set of statistics is shown here:
+
+```
+Loop time of 1.76553 on 1 procs for 100 steps with 32000 atoms
+
+Performance: 24468.549 tau/day, 56.640 timesteps/s
+100.0% CPU use with 1 MPI tasks x 1 OpenMP threads
+
+MPI task timing breakdown:
+Section |  min time  |  avg time  |  max time  |%varavg| %total
+---------------------------------------------------------------
+Pair    | 1.5244     | 1.5244     | 1.5244     |   0.0 | 86.34
+Neigh   | 0.19543    | 0.19543    | 0.19543    |   0.0 | 11.07
+Comm    | 0.016556   | 0.016556   | 0.016556   |   0.0 |  0.94
+Output  | 7.2241e-05 | 7.2241e-05 | 7.2241e-05 |   0.0 |  0.00
+Modify  | 0.023852   | 0.023852   | 0.023852   |   0.0 |  1.35
+Other   |            | 0.005199   |            |       |  0.29
+
+Nlocal:    32000 ave 32000 max 32000 min
+Histogram: 1 0 0 0 0 0 0 0 0 0
+Nghost:    19657 ave 19657 max 19657 min
+Histogram: 1 0 0 0 0 0 0 0 0 0
+Neighs:    1.20283e+06 ave 1.20283e+06 max 1.20283e+06 min
+Histogram: 1 0 0 0 0 0 0 0 0 0
+
+Total # of neighbors = 1202833
+Ave neighs/atom = 37.5885
+Neighbor list builds = 5
+Dangerous builds not checked
+Total wall time: 0:00:01
+```
+{: .log.lammps}
+
+## Useful keyword to search for
+  * The ```Loop time``` is the total wall-clock time for the simulation to run. (source: LAMMPS manual)
+Use the following to extract this from ```log.lammps```:
+```
+grep "Loop time" log.lammps| sed 's/|/ /' | awk '{print $4}'
```

Comment:
This is potentially a really complicated/intimidating command line for people who are not really familiar with bash. While it is convenient, is it really necessary? We can just highlight the number in the output